{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Listening ...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\loop.py\", line 60, in run_forever\n",
      "    allowed_updates=allowed_updates)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 1000, in getUpdates\n",
      "    return self._api_request('getUpdates', _rectify(p))\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 491, in _api_request\n",
      "    return api.request((self._token, method, params, files), **kwargs)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 155, in request\n",
      "    return _parse(r)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 150, in _parse\n",
      "    raise exception.TelegramError(description, error_code, data)\n",
      "telepot.exception.TelegramError: ('Conflict: terminated by other getUpdates request; make sure that only one bot instance is running', 409, {'ok': False, 'error_code': 409, 'description': 'Conflict: terminated by other getUpdates request; make sure that only one bot instance is running'})\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\loop.py\", line 60, in run_forever\n",
      "    allowed_updates=allowed_updates)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 1000, in getUpdates\n",
      "    return self._api_request('getUpdates', _rectify(p))\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 491, in _api_request\n",
      "    return api.request((self._token, method, params, files), **kwargs)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 155, in request\n",
      "    return _parse(r)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 150, in _parse\n",
      "    raise exception.TelegramError(description, error_code, data)\n",
      "telepot.exception.TelegramError: ('Conflict: terminated by other getUpdates request; make sure that only one bot instance is running', 409, {'ok': False, 'error_code': 409, 'description': 'Conflict: terminated by other getUpdates request; make sure that only one bot instance is running'})\n"
     ]
    }
   ],
   "source": [
    "\n",
    "# coding: utf-8\n",
    "\n",
    "# In[1]:\n",
    "\n",
    "\n",
    "import telepot\n",
    "from telepot.loop import MessageLoop\n",
    "from telepot.namedtuple import (\n",
    "    ReplyKeyboardMarkup, \n",
    "    KeyboardButton,\n",
    "    InlineKeyboardMarkup,\n",
    "    InlineKeyboardButton\n",
    ")\n",
    "from random import choice\n",
    "import json\n",
    "\n",
    "\n",
    "# In[2]:\n",
    "\n",
    "#自行更新  #自行更新   #自行更新\n",
    "TOKEN = '563854488:AAGOZLDjlWf7jRz869DUVwRzPwsGcfJhYXw' #自行更新\n",
    "password = '9487'                                           #自行更新\n",
    "#自行更新  #自行更新   #自行更新\n",
    "\n",
    "\n",
    "# In[3]:\n",
    "\n",
    "\n",
    "bot = telepot.Bot(TOKEN)\n",
    "\n",
    "# task[number<int>] = { ans:<string>, score<:int>, }\n",
    "task = {}\n",
    "# self[username<sting>] = {team<int>, score:<int>, }\n",
    "self = {}\n",
    "# team[team_number<int> = {members<list>, total<int>, task:[<int finish<bool>]}, bounus:<int>, }\n",
    "team = {1:{\n",
    "          \"task_number\" : False, \n",
    "          \"members\":[], \n",
    "          \"total\":0},\n",
    "        2:{},\n",
    "        }\n",
    "\n",
    "# 列印訊息接收log\n",
    "def print_msg(msg):\n",
    "    print(json.dumps(msg, indent=4))\n",
    "\n",
    "# 接收chat後執行：\n",
    "def on_chat(msg):\n",
    "    # 得取基礎資料：訊息類型\\聊天室總類\\聊天室id\n",
    "    header = telepot.glance(msg, flavor=\"chat\")\n",
    "    print_msg(msg)\n",
    "    data=\"\"\n",
    "    if header[0] == \"text\":\n",
    "        text = msg[\"text\"]\n",
    "        username = msg[\"from\"][\"username\"]\n",
    "        try:\n",
    "            userteam = self[username][team]\n",
    "        except:\n",
    "            pass\n",
    "        # command\n",
    "        if text.startswith(\"/\"):\n",
    "            command = text.lstrip(\"/\")\n",
    "            \n",
    "            if command == \"start\":\n",
    "                text = \"OK， {}\\n你準備好了...... 讓我們開始一場奇幻冒險ㄅ OwO\"\n",
    "                bot.sendMessage(header[2], text.format(msg[\"from\"][\"first_name\"]))\n",
    "                bot.sendMessage(header[2], \"請先輸入您的組別 /team <阿拉伯數字>～\\n請小心輸入，這會影響你的計分唷～\")\n",
    "                bot.sendMessage(header[2], \"/list查看任務紀錄\\n/total 獲取總計分\\ntask<題號><答案> 回傳任務答案\\n\")\n",
    "\n",
    "\n",
    "            elif command[:4] == \"team\":\n",
    "                data = command[4:].split()\n",
    "                data=int(data[0])\n",
    "                if username in self:\n",
    "                    if username in team[data][\"members\"]:\n",
    "                        bot.sendMessage(header[2], \"ni已經註冊過ㄖ\")\n",
    "                    else:\n",
    "                        team[self[username][team]].remove(username)\n",
    "                        team[data]['members'].append(username)\n",
    "                else:\n",
    "                    self[username] = {\"team\" : data}\n",
    "                    team[data]['members'].append(username)\n",
    "                    \n",
    "            elif command[:3] == \"add\":\n",
    "                data = command[3:].split()\n",
    "                if len(data) == 4:\n",
    "                    if data[0] == password:\n",
    "                        task[data[1]][\"ans\"] = data[2]\n",
    "                        task[data[1]][\"score\"] = data[3]\n",
    "                        bot.sendMessage(header[2], \"Thank you! We have add it to our data base\")\n",
    "\n",
    "                    else:\n",
    "                        bot.sendMessage(header[2], \"Please enter the true password !\")\n",
    "                else:\n",
    "                    bot.sendMessage(header[2], \"Please enter the correct data!\\n/add <password> <task_number> <task_answer> <task_score> !\")\n",
    "            # user answer the question                     \n",
    "            elif command[:4] == \"task\":\n",
    "                [print(task[i][score]) for i in task]\n",
    "            elif command[:6] == \"status\":\n",
    "                data = commacd[4:].split()\n",
    "                task_number = data[0]\n",
    "                task_answer = data[1]\n",
    "                if task[task_number] == task_answer:\n",
    "                    if team[userteam][task_number] == True:\n",
    "                        bot.sendMessage(header[2], \"You have sended the correct answer ! go on to find the next task\")\n",
    "                    else:\n",
    "                        team[userteam][task_number] = True\n",
    "                        team[userteam][total] += task[score]\n",
    "            # user request the list of team info\n",
    "            elif command[:4] == \"list\":\n",
    "                for i in team[userteam]:\n",
    "                    if i == \"members\":\n",
    "                        for j in i:\n",
    "                            print(j, end=\" \")\n",
    "                        print(\"\")\n",
    "                    else:\n",
    "                        print(i)\n",
    "                                     \n",
    "            elif command[:5] == \"total\":\n",
    "                bot.sendMessage(header[2],team[userteam][total])\n",
    "\n",
    "MessageLoop(bot, {\n",
    "    'chat': on_chat,\n",
    "    #'callback_query': on_callback_query,\n",
    "}).run_as_thread()\n",
    "\n",
    "print('Listening ...')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{}"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\loop.py\", line 60, in run_forever\n",
      "    allowed_updates=allowed_updates)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 1000, in getUpdates\n",
      "    return self._api_request('getUpdates', _rectify(p))\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 491, in _api_request\n",
      "    return api.request((self._token, method, params, files), **kwargs)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 155, in request\n",
      "    return _parse(r)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 150, in _parse\n",
      "    raise exception.TelegramError(description, error_code, data)\n",
      "telepot.exception.TelegramError: ('Conflict: terminated by other getUpdates request; make sure that only one bot instance is running', 409, {'ok': False, 'error_code': 409, 'description': 'Conflict: terminated by other getUpdates request; make sure that only one bot instance is running'})\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\loop.py\", line 60, in run_forever\n",
      "    allowed_updates=allowed_updates)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 1000, in getUpdates\n",
      "    return self._api_request('getUpdates', _rectify(p))\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 491, in _api_request\n",
      "    return api.request((self._token, method, params, files), **kwargs)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 155, in request\n",
      "    return _parse(r)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 150, in _parse\n",
      "    raise exception.TelegramError(description, error_code, data)\n",
      "telepot.exception.TelegramError: ('Conflict: terminated by other getUpdates request; make sure that only one bot instance is running', 409, {'ok': False, 'error_code': 409, 'description': 'Conflict: terminated by other getUpdates request; make sure that only one bot instance is running'})\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\loop.py\", line 60, in run_forever\n",
      "    allowed_updates=allowed_updates)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 1000, in getUpdates\n",
      "    return self._api_request('getUpdates', _rectify(p))\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 491, in _api_request\n",
      "    return api.request((self._token, method, params, files), **kwargs)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 155, in request\n",
      "    return _parse(r)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 150, in _parse\n",
      "    raise exception.TelegramError(description, error_code, data)\n",
      "telepot.exception.TelegramError: ('Conflict: terminated by other getUpdates request; make sure that only one bot instance is running', 409, {'ok': False, 'error_code': 409, 'description': 'Conflict: terminated by other getUpdates request; make sure that only one bot instance is running'})\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\loop.py\", line 60, in run_forever\n",
      "    allowed_updates=allowed_updates)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 1000, in getUpdates\n",
      "    return self._api_request('getUpdates', _rectify(p))\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 491, in _api_request\n",
      "    return api.request((self._token, method, params, files), **kwargs)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 155, in request\n",
      "    return _parse(r)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 150, in _parse\n",
      "    raise exception.TelegramError(description, error_code, data)\n",
      "telepot.exception.TelegramError: ('Conflict: terminated by other getUpdates request; make sure that only one bot instance is running', 409, {'ok': False, 'error_code': 409, 'description': 'Conflict: terminated by other getUpdates request; make sure that only one bot instance is running'})\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\loop.py\", line 60, in run_forever\n",
      "    allowed_updates=allowed_updates)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 1000, in getUpdates\n",
      "    return self._api_request('getUpdates', _rectify(p))\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 491, in _api_request\n",
      "    return api.request((self._token, method, params, files), **kwargs)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 155, in request\n",
      "    return _parse(r)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 150, in _parse\n",
      "    raise exception.TelegramError(description, error_code, data)\n",
      "telepot.exception.TelegramError: ('Conflict: terminated by other getUpdates request; make sure that only one bot instance is running', 409, {'ok': False, 'error_code': 409, 'description': 'Conflict: terminated by other getUpdates request; make sure that only one bot instance is running'})\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\loop.py\", line 60, in run_forever\n",
      "    allowed_updates=allowed_updates)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 1000, in getUpdates\n",
      "    return self._api_request('getUpdates', _rectify(p))\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 491, in _api_request\n",
      "    return api.request((self._token, method, params, files), **kwargs)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 155, in request\n",
      "    return _parse(r)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 150, in _parse\n",
      "    raise exception.TelegramError(description, error_code, data)\n",
      "telepot.exception.TelegramError: ('Conflict: terminated by other getUpdates request; make sure that only one bot instance is running', 409, {'ok': False, 'error_code': 409, 'description': 'Conflict: terminated by other getUpdates request; make sure that only one bot instance is running'})\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\loop.py\", line 60, in run_forever\n",
      "    allowed_updates=allowed_updates)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 1000, in getUpdates\n",
      "    return self._api_request('getUpdates', _rectify(p))\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 491, in _api_request\n",
      "    return api.request((self._token, method, params, files), **kwargs)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 155, in request\n",
      "    return _parse(r)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 150, in _parse\n",
      "    raise exception.TelegramError(description, error_code, data)\n",
      "telepot.exception.TelegramError: ('Conflict: terminated by other getUpdates request; make sure that only one bot instance is running', 409, {'ok': False, 'error_code': 409, 'description': 'Conflict: terminated by other getUpdates request; make sure that only one bot instance is running'})\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\loop.py\", line 60, in run_forever\n",
      "    allowed_updates=allowed_updates)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 1000, in getUpdates\n",
      "    return self._api_request('getUpdates', _rectify(p))\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 491, in _api_request\n",
      "    return api.request((self._token, method, params, files), **kwargs)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 155, in request\n",
      "    return _parse(r)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 150, in _parse\n",
      "    raise exception.TelegramError(description, error_code, data)\n",
      "telepot.exception.TelegramError: ('Conflict: terminated by other getUpdates request; make sure that only one bot instance is running', 409, {'ok': False, 'error_code': 409, 'description': 'Conflict: terminated by other getUpdates request; make sure that only one bot instance is running'})\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\loop.py\", line 60, in run_forever\n",
      "    allowed_updates=allowed_updates)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 1000, in getUpdates\n",
      "    return self._api_request('getUpdates', _rectify(p))\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 491, in _api_request\n",
      "    return api.request((self._token, method, params, files), **kwargs)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 155, in request\n",
      "    return _parse(r)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 150, in _parse\n",
      "    raise exception.TelegramError(description, error_code, data)\n",
      "telepot.exception.TelegramError: ('Conflict: terminated by other getUpdates request; make sure that only one bot instance is running', 409, {'ok': False, 'error_code': 409, 'description': 'Conflict: terminated by other getUpdates request; make sure that only one bot instance is running'})\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\loop.py\", line 60, in run_forever\n",
      "    allowed_updates=allowed_updates)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 1000, in getUpdates\n",
      "    return self._api_request('getUpdates', _rectify(p))\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 491, in _api_request\n",
      "    return api.request((self._token, method, params, files), **kwargs)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 155, in request\n",
      "    return _parse(r)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 150, in _parse\n",
      "    raise exception.TelegramError(description, error_code, data)\n",
      "telepot.exception.TelegramError: ('Conflict: terminated by other getUpdates request; make sure that only one bot instance is running', 409, {'ok': False, 'error_code': 409, 'description': 'Conflict: terminated by other getUpdates request; make sure that only one bot instance is running'})\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\loop.py\", line 60, in run_forever\n",
      "    allowed_updates=allowed_updates)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 1000, in getUpdates\n",
      "    return self._api_request('getUpdates', _rectify(p))\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 491, in _api_request\n",
      "    return api.request((self._token, method, params, files), **kwargs)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 155, in request\n",
      "    return _parse(r)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 150, in _parse\n",
      "    raise exception.TelegramError(description, error_code, data)\n",
      "telepot.exception.TelegramError: ('Conflict: terminated by other getUpdates request; make sure that only one bot instance is running', 409, {'ok': False, 'error_code': 409, 'description': 'Conflict: terminated by other getUpdates request; make sure that only one bot instance is running'})\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\loop.py\", line 60, in run_forever\n",
      "    allowed_updates=allowed_updates)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 1000, in getUpdates\n",
      "    return self._api_request('getUpdates', _rectify(p))\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 491, in _api_request\n",
      "    return api.request((self._token, method, params, files), **kwargs)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 155, in request\n",
      "    return _parse(r)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 150, in _parse\n",
      "    raise exception.TelegramError(description, error_code, data)\n",
      "telepot.exception.TelegramError: ('Conflict: terminated by other getUpdates request; make sure that only one bot instance is running', 409, {'ok': False, 'error_code': 409, 'description': 'Conflict: terminated by other getUpdates request; make sure that only one bot instance is running'})\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\loop.py\", line 60, in run_forever\n",
      "    allowed_updates=allowed_updates)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 1000, in getUpdates\n",
      "    return self._api_request('getUpdates', _rectify(p))\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 491, in _api_request\n",
      "    return api.request((self._token, method, params, files), **kwargs)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 155, in request\n",
      "    return _parse(r)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 150, in _parse\n",
      "    raise exception.TelegramError(description, error_code, data)\n",
      "telepot.exception.TelegramError: ('Conflict: terminated by other getUpdates request; make sure that only one bot instance is running', 409, {'ok': False, 'error_code': 409, 'description': 'Conflict: terminated by other getUpdates request; make sure that only one bot instance is running'})\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\loop.py\", line 60, in run_forever\n",
      "    allowed_updates=allowed_updates)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 1000, in getUpdates\n",
      "    return self._api_request('getUpdates', _rectify(p))\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\__init__.py\", line 491, in _api_request\n",
      "    return api.request((self._token, method, params, files), **kwargs)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 155, in request\n",
      "    return _parse(r)\n",
      "  File \"C:\\Users\\User\\Anaconda3\\lib\\site-packages\\telepot\\api.py\", line 150, in _parse\n",
      "    raise exception.TelegramError(description, error_code, data)\n",
      "telepot.exception.TelegramError: ('Conflict: terminated by other getUpdates request; make sure that only one bot instance is running', 409, {'ok': False, 'error_code': 409, 'description': 'Conflict: terminated by other getUpdates request; make sure that only one bot instance is running'})\n"
     ]
    }
   ],
   "source": [
    "task"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "admins_list=[]\n",
    "admins_dict=telBot.getChatAdministrators (msg[\"chat\"][\"id\"])\n",
    "chat_name=msg[\"chat\"][\"title\"].encode('utf8')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
